packages/blockchain-info-components/index.d.ts

```diff
@@ -1,30 +1,138 @@
-import { StatelessComponent, CSSProperties } from 'react'
+import { StatelessComponent, PureComponent, CSSProperties } from 'react'
 import { DefaultTheme } from 'styled-components'
+import { IcoMoonType } from './src/Icons/Icomoon'
+// import { ImageType } from './src/Images/Images'
+
+export const BlockchainLoader: StatelessComponent<{
+  width?: string
+  height?: string
+}>
+export const Button: StatelessComponent<{
+  nature?:
+    | 'copy'
+    | 'dark'
+    | 'empty-secondary'
+    | 'empty'
+    | 'gray-3'
+    | 'green'
+    | 'light'
+    | 'primary'
+    | 'purple'
+    | 'received'
+    | 'secondary'
+    | 'sent'
+    | 'success'
+    | 'transferred'
+    | 'warning'
+    | 'white-transparent'
+  fullwidth?: boolean
+  disabled?: boolean
+  rounded?: boolean
+  bold?: boolean
+  small?: boolean
+  uppercase?: boolean
+  capitalize?: boolean
+  width?: string
+  padding?: string
+  margin?: string
+  jumbo?: boolean
+  height?: string
+  className?: string
+  onClick?: () => void
+  style?: object
+}>
 
-export function BlockchainLoader(...args: any): any
-export function Button(...args: any): any
 export function ComponentDropdown(...args): any
-export function FontGlobalStyles(...args: any): any
-export function Icon(...args: any): any
-export function IconGlobalStyles(...args: any): any
-export function Image(...args: any): any
-export function Link(...args: any): any
-export function Modal(...args: any): any
-export function ModalBody(...args: any): any
-export function ModalHeader(...args: any): any
+// export const ComponentDropdown: PureComponent<{
+//   color?: string
+//   toggleOnClassback?: boolean
+//   opened?: boolean
+//   uppercase?: boolean
+//   down?: boolean
+//   forceSelected?: boolean
+//   selectedComponent:
+//   onClick?: () => void
+//   callback?: () => void
+// }>
+
+export const FontGlobalStyles: StatelessComponent<{}>
+export const Icon: StatelessComponent<{
+  name?: keyof IcoMoonType
+  weight?: number
+  size?: string
+  cursor?: boolean
+  color?: string
+  style?: object
+  onClick?: () => void
+}>
+export const IconGlobalStyles: StatelessComponent<{}>
+export const Image: StatelessComponent<{
+  name?: string
```

Comment:
I think name is required, so no `?`

Reply:
And why aren't you using the ImagesType for name?

Reply:
Converting the images.js file into a .ts file causes all the image imports to error because TS doesn't recognize it. I did find a solution here, https://stackoverflow.com/questions/51100401/typescript-image-import, adding `declare module "*.png"`, etc to a d.ts file. I can work through this now, if you think it's the right solution.

Reply:
Yeah if that works lets do that please

Reply:
Done